src/Renderer/LayeredMaterial.js

```diff
@@ -362,29 +362,29 @@ LayeredMaterial.prototype.getLoadedTexturesCount = function getLoadedTexturesCou
     return this.loadedTexturesCount[l_ELEVATION] + this.loadedTexturesCount[l_COLOR];
 };
 
-LayeredMaterial.prototype.isColorLayerDownscaled = function isColorLayerDownscaled(layer, level) {
-    return this.textures[l_COLOR][this.getLayerTextureOffset(layer)].level < level;
+LayeredMaterial.prototype.isColorLayerDownscaled = function isColorLayerDownscaled(layerId, level) {
+    if (this.textures[l_COLOR][this.getLayerTextureOffset(layerId)]) {
+        return this.textures[l_COLOR][this.getLayerTextureOffset(layerId)].coordWMTS.zoom < level;
+    } else {
+        return false;
+    }
 };
 
 LayeredMaterial.prototype.isLayerTypeDownscaled = function isLayerTypeDownscaled(layerType, level) {
     if (layerType === l_ELEVATION) {
         const tex = this.textures[l_ELEVATION][0];
-        // 3 possible cases
-        //   - initialization (no texture)
-        if (tex === undefined) {
-            return true;
-        }
         //   - blank texture (eg: empty xbil texture)
-        if (tex.level === EMPTY_TEXTURE_LEVEL) {
+        if (tex.coordWMTS.zoom === EMPTY_TEXTURE_ZOOM) {
             return false;
         }
         //   - regular texture
-        return tex.level < level;
+        return tex.coordWMTS.zoom < level;
     } else if (layerType === l_COLOR) {
         // browse each layer
         for (let index = 0, max = this.colorLayersId.length; index < max; index++) {
             const offset = this.getTextureOffsetByLayerIndex(index);
-            if (this.textures[l_COLOR][offset].level < level) {
+            const texture = this.textures[l_COLOR][offset];
+            if (texture.coordWMTS.zoom < level) {
                 return true;
```

Comment:
Ok ok yes, I miread the diff in my second comment. I thought that's what you meant by 

> I don't think because we must check all the textures

So we're back at my first comment, which is actually correct: 

> nits :
> ```js
> return this.textures[l_color][offset].coordWMTS.zoom < level;
> ```

And 

> `if (condition) return true` is always equivalent to `return condition;`

Do you see what I mean?